src/Stryker.Core/Stryker.Core/Options/StrykerOptions.cs

```diff
@@ -388,13 +388,11 @@ private IEnumerable<FilePattern> ValidateDiffIgnoreFiles(IEnumerable<string> dif
             return mappedDiffIgnoreFiles;
         }
 
-        private static IEnumerable<Regex> ValidateIgnoredMethods(IEnumerable<string> methodPatterns)
-        {
-            foreach (var methodPattern in methodPatterns.Where(x => !string.IsNullOrEmpty(x)))
-            {
-                yield return new Regex("^" + Regex.Escape(methodPattern).Replace("\\*", ".*") + "$", RegexOptions.IgnoreCase);
-            }
-        }
+        private static IEnumerable<Regex> ValidateIgnoredMethods(IEnumerable<string> methodPatterns) =>
+            methodPatterns
+                .Where(x => !string.IsNullOrEmpty(x))
+                .Select(methodPattern => new Regex("^(?:[^.]*\\.)*" + Regex.Escape(methodPattern).Replace("\\*", "[^.]*") + "$", RegexOptions.IgnoreCase))
```

Comment:
As with all regexes, it could be rewritten using only string members. It would be a lot more code.

Reply:
Sure. I mean, I suggest either using string members or adding some comments - whatever makes things more clear here for you and others :)

Reply:
Or we can ask some bypasser if that's clear :D @richardwerkman?

Reply:
Indeed this could use some explaination in a comment. Why does the regex start with `^(?:[^.]*\\.)*`?

Also, this validate is now tested together with the filter I see. But we are working on a rework of how the options work. Could you add seperate tests for this validation? That also helps as explaination of course. And will make my life a lot easier when merging with the refactor :)

Reply:
Comment and validation tests have both been added. 😀